src/modules/ekf2/EKF/height_fusion.cpp

```diff
@@ -40,79 +40,80 @@
 
 void Ekf::fuseBaroHgt()
 {
-	Vector2f baro_hgt_innov_gate;
-	Vector3f baro_hgt_obs_var;
-
 	// vertical position innovation - baro measurement has opposite sign to earth z axis
 	const float unbiased_baro = _baro_sample_delayed.hgt - _baro_b_est.getBias();
-	_baro_hgt_innov(2) = _state.pos(2) + unbiased_baro - _baro_hgt_offset;
-	// observation variance - user parameter defined
-	baro_hgt_obs_var(2) = sq(fmaxf(_params.baro_noise, 0.01f));
-	// innovation gate size
-	baro_hgt_innov_gate(1) = fmaxf(_params.baro_innov_gate, 1.0f);
+
+	_baro_hgt_innov = _state.pos(2) + unbiased_baro - _baro_hgt_offset;
 
 	// Compensate for positive static pressure transients (negative vertical position innovations)
 	// caused by rotor wash ground interaction by applying a temporary deadzone to baro innovations.
 	const float deadzone_start = 0.0f;
 	const float deadzone_end = deadzone_start + _params.gnd_effect_deadzone;
 
 	if (_control_status.flags.gnd_effect) {
-		if (_baro_hgt_innov(2) < -deadzone_start) {
-			if (_baro_hgt_innov(2) <= -deadzone_end) {
-				_baro_hgt_innov(2) += deadzone_end;
+		if (_baro_hgt_innov < -deadzone_start) {
+			if (_baro_hgt_innov <= -deadzone_end) {
+				_baro_hgt_innov += deadzone_end;
```

Comment:
Somewhat unrelated, why isn't this a constraint within the deadzone?

```suggestion
				_baro_hgt_innov = math::constrain(_baro_hgt_innov, -deadzone_end, -deadzone_start);
```

Reply:
It's just to get a continuous behavior:
- if below the `deadzone_end`: force to `deadzone_start`
- if above the `deadzone_end`: remove `deadzone_end` to start at zero

Reply:
Thanks.